Sources/NIO/Bootstrap.swift

```diff
@@ -649,6 +794,57 @@ public final class ClientBootstrap: NIOClientTCPBootstrapProtocol {
             return eventLoop.submit(setupChannel).flatMap { $0 }
         }
     }
+    
+    /// A channel option which can be applied to bootstrap using shorthand notation.
+    /// - See: ClientBootstrap.clientOptions(_ options: [ShorthandClientBootstrapOption])
+    public struct ShorthandClientBootstrapOption {
+        private let data: ShorthandClientOption
+        
+        private init(_ data: ShorthandClientOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied ClientBootstrap
+        /// - Parameter to: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        func applyOption(to clientBootstrap: ClientBootstrap) {
+            data.applyOption(to: clientBootstrap)
+        }
+        
+        fileprivate enum ShorthandClientOption {
+            case reuseAddr
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to clientBootstrap: ClientBootstrap) {
+                switch self {
+                case .reuseAddr:
+                    _ = clientBootstrap.channelOption(ChannelOptions.socketOption(.reuseaddr), value: 1)
+                case .allowRemoteHalfClosure(let value):
+                    _ = clientBootstrap.channelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension ClientBootstrap.ShorthandClientBootstrapOption: Hashable {}
```

Comment:
now that the type is nested, `ClientBootstrap` is duplicated. What about

    ClientBootstrap.ConvenienceOption

or do folks (@Lukasa / @glbrntt / @Davidde94 / @PeterAdams-A) prefer "shorthand" over "convenience"?

Reply:
Do we need a prefix; `ClientBootstrap.Option`?

Reply:
you're right, we don't, I like `ClientBootstrap.Option`.

Reply:
Is just option slightly misleading?  We don't support all options via this method.

Reply:
I'll go with option for now as you both liked that.

Reply:
ServerOption and ChildOption on ServerBootstrap, otherwise just Option - done.